modules/convert/convert.go

```diff
@@ -30,28 +30,76 @@ func ToEmail(email *models.EmailAddress) *api.Email {
+			Protected:                     false,
+			RequiredApprovals:             0,
+			EnableStatusCheck:             false,
+			StatusCheckContexts:           []string{},
+			UserCanPush:                   true,
+			UserCanMerge:                  true,
+			EffectiveBranchProtectionName: "",
+			EffectiveBRanchProtectionID:   0,
 		}
 	}
+	branchProtectionName := ""
+	var branchProtectionID int64
+	if isRepoAdmin {
+		branchProtectionName = bp.BranchName
+		branchProtectionID = bp.ID
+	}
+
 	return &api.Branch{
-		Name:                b.Name,
-		Commit:              ToCommit(repo, c),
-		Protected:           true,
-		RequiredApprovals:   bp.RequiredApprovals,
-		EnableStatusCheck:   bp.EnableStatusCheck,
-		StatusCheckContexts: bp.StatusCheckContexts,
-		UserCanPush:         bp.CanUserPush(user.ID),
-		UserCanMerge:        bp.CanUserMerge(user.ID),
+		Name:                          b.Name,
+		Commit:                        ToCommit(repo, c),
+		Protected:                     true,
+		RequiredApprovals:             bp.RequiredApprovals,
+		EnableStatusCheck:             bp.EnableStatusCheck,
+		StatusCheckContexts:           bp.StatusCheckContexts,
+		UserCanPush:                   bp.CanUserPush(user.ID),
+		UserCanMerge:                  bp.CanUserMerge(user.ID),
+		EffectiveBranchProtectionName: branchProtectionName,
+		EffectiveBRanchProtectionID:   branchProtectionID,
+	}
+}
+
+// ToBranchProtection convert a ProtectedBranch to api.BranchProtection
+func ToBranchProtection(bp *models.ProtectedBranch) *api.BranchProtection {
+	pushWhitelistUsernames, err := models.GetUsernamesByIDs(bp.WhitelistUserIDs)
+	if err != nil {
+		log.Error("GetUsernamesByIDs (WhitelistUserIDs): %v", err)
+	}
+	mergeWhitelistUsernames, err := models.GetUsernamesByIDs(bp.MergeWhitelistUserIDs)
+	if err != nil {
+		log.Error("GetUsernamesByIDs (MergeWhitelistUserIDs): %v", err)
+	}
+	approvalsWhitelistUsernames, err := models.GetUsernamesByIDs(bp.ApprovalsWhitelistUserIDs)
+	if err != nil {
+		log.Error("GetUsernamesByIDs (ApprovalsWhitelistUserIDs): %v", err)
+	}
+
+	return &api.BranchProtection{
+		ID:                          bp.ID,
+		BranchName:                  bp.BranchName,
+		EnablePush:                  bp.CanPush,
+		EnablePushWhitelist:         bp.EnableWhitelist,
+		PushWhitelistUsernames:      pushWhitelistUsernames,
+		PushWhitelistTeamIDs:        bp.WhitelistTeamIDs,
+		PushWhitelistDeployKeys:     bp.WhitelistDeployKeys,
+		EnableMergeWhitelist:        bp.EnableMergeWhitelist,
+		MergeWhitelistUsernames:     mergeWhitelistUsernames,
+		MergeWhitelistTeamIDs:       bp.MergeWhitelistTeamIDs,
+		EnableStatusCheck:           bp.EnableStatusCheck,
+		StatusCheckContexts:         bp.StatusCheckContexts,
+		RequiredApprovals:           bp.RequiredApprovals,
+		EnableApprovalsWhitelist:    bp.EnableApprovalsWhitelist,
+		ApprovalsWhitelistUsernames: approvalsWhitelistUsernames,
+		ApprovalsWhitelistTeamIDs:   bp.ApprovalsWhitelistTeamIDs,
```

Comment:
IMHO, teams should be informed by name, not ID.

Reply:
I choosed it because existing APIs use {username} to access user information while {id} is used to access team information, so I thought it made sense. For information returned maybe it should return the complete user and team API object?

Reply:
I don't know. I try to avoid giving too many opinions about APIs because I'm not an API user, however, seeing this API in isolation team ids don't look right. I'd like to hear the opinions of other maintainers about this if possible.

Reply:
looked at the api ... and the only good solutions are to return id or the whole struct

Reply:
> 
> 
> looked at the api ... and the only good solutions are to return id or the whole struct

Is there a reason teams can't be specified by name?

Reply:
When it comes to setting the value (post/editing) I already use team id to do other things like adding repositories to a team, so for me it makes sense to continue the same way. Returning the whole struct seem fine for me, if it is not considered too resource ineffective.